airflow/cli/commands/variable_command.py

```diff
@@ -86,7 +87,16 @@ def variables_import(args):
         except JSONDecodeError:
             raise SystemExit("Invalid variables file.")
     suc_count = fail_count = 0
+    skipped = set()
+    action_on_existing = args.action_on_existing_key
     for k, v in var_json.items():
+        if action_on_existing != "overwrite":
+            exists_q = session.scalar(select(Variable).where(Variable.key == k))
```

Comment:
This could be a bit expensive. Something like this could be better

```python
existing_ids = set(session.scalars(select(Variable.key).where(Variable.key.in_(var_json))))
if existing_ids and action_on_existing == "fail":
    # Fail, existing variable keys found.
```